blog/content/post/2021-02-21-amd-milan-osf.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Open Source Firmware on AMD Milan server processors
+abstract: 'There were times where AMD was actively supporting open source
+           firmware ecosystem by providing silicon initialization code. With
+           a few years break AMD is now trying to support open source firmware
+           again with the mobile platform like Chromebooks. However, the recent
+           achievements have made everybody think that open source firmware is
+           also possible on severs.'
+cover: /covers/amd-epyc.jpg
+author: michal.zygowski
+layout: post
+published: true
+date: 2021-02-21
+archives: "2021"
+
+tags:
+  - firmware
+categories:
+  - Firmware
+
+---
+
+# Introduction
+
+Do you remember times when AMD was actively supporting open source firmware?
+Back in the 2008 AMD was delivering the AGESA silicon initialization code and
+was contributing it to coreboot. Up to 2014 AMD was release the AGESA code in
+public. These were the golden years of open source firmware for AMD processors
+and many platforms emerged in coreboot. You may know those platform under
+various names like PC Engines ALIX and APU1. They benefit still from fully open
+source implementation. However since 2014 AMD started to deliver the AGESA in
+binary form only. And that situation is maintained till today with some
+exceptions (after AMD Steppe Eagle there were no AGESA binary releases
+anymore). Google encouraged the AMD to work on FSP compatible interface for
+AGESA for their Chromebooks and today we have a AMD Picasso support in
+coreboot. If you want to know more about Open Source Firmware on AMD history,
+please refer to our FOSDEM presentations:
+
+* https://video.fosdem.org/2020/K.4.401/coreboot_amd.mp4
+* https://video.fosdem.org/2021/D.firmware/firmware_osfsoap2.mp4
+
+Everybody hopes that AMD server will get open source firmware too. According to
+rumors something is moving into that direction. However, in this blog post I
+will describe the real current situation.
```

Comment:
@miczyg1 any pointers to those rumours?

Reply:
No pointers unfortunately

Reply:
Let's avoid claims that have no official public proofs, just to avoid any misunderstanding with AMD.

Reply:
https://9esec.io/blog/the-future-of-open-source-firmware-on-server-systems/
https://www.phoronix.com/scan.php?page=news_item&px=AMD-OpenBMC-2020-Progress


Actually it is you who said something si going on : https://www.phoronix.com/scan.php?page=news_item&px=Open-Source-AMD-FW-2021
http://mirroronet.pl/pub/mirrors/video.fosdem.org/2021/D.firmware/firmware_osfsoap2.mp4 at 12:05

Reply:
@miczyg1 that information rely on coreboot leadership meeting minutes 2020: https://docs.google.com/document/d/1sYmonZnZxZ-CA49clOjNs7zewR_2Mdm9eMfxka0v5i0/edit?usp=sharing

Please search for AMD

Reply:
Not much there:
```
When AMD gets to server chips, they should engage since there’s probably overlap through FSP
```

Reply:
Added it to the post

Reply:
> AMD: Apparently Zen scales up and down so the Zen mobile code might be reusable for server stuff (multi-socket might take some work)

Reply:
Added the second cite as well